common/src/com/thoughtworks/go/domain/materials/Modifications.java

```diff
@@ -129,6 +129,11 @@ public boolean shouldBeIgnoredByFilterIn(MaterialConfig materialConfig) {
             return false;
         }
         Set<ModifiedFile> allFiles = getAllFiles(this);
+
+        if (allFiles.isEmpty()) {
```

Comment:
hmm..ok lets discuss more about this. closing the PR & creating a new one with just the rails change.